tests/test_pytypes.py

```diff
@@ -488,3 +488,45 @@ def test_builtin_functions():
         "object of type 'generator' has no len()",
         "'generator' has no length",
     ]  # PyPy
+
+
+def test_weakref():
+    from weakref import getweakrefcount
+
+    # Apparently, you cannot weakly reference an object()
+    class WeaklyReferenced(object):
+        pass
+
+    called = [0]
+
+    def callback(wr):
+        # No `nonlocal` in Python 2
+        called[0] += 1
```

Comment:
Why not just use `global`?

Reply:
Because it's not `global`? It's `nonlocal`, I think (?), but that's not available in Python 2?

Reply:
I thought `global` meant "first" variable of `name` going from the global scope towards the current scope. Carry on.

Reply:
I wasn't completely sure either, but this test says I'm right (beware, this is ugly!):

```
>>> def f():
...     i = 0
...     def g():
...             global i
...             i += 1
...     g()
...     g()
...     return i
... 
>>> f()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 6, in f
  File "<stdin>", line 5, in g
NameError: name 'i' is not defined
>>> i = 0
>>> f()
0
>>> i
2
```


But wait, I can still pick something better!

Reply:
Better this way? It's just as weird, but looks better and slightly more descriptive.